aten/src/ATen/native/xnnpack/Convolution.cpp

```diff
@@ -154,7 +155,9 @@ ContextConv2D create(
       weight_nhwc.size(Layout::Filter::input) * groups,               // input_pixel_stride
       weight_nhwc.size(Layout::Filter::output),                       // output_pixel_stride
       weight_nhwc.data_ptr<float>(),                                  // kernel
-      (bias && bias->defined()) ? bias->data_ptr<float>() : nullptr,  // bias
+      (bias && bias->defined())
+          ? bias->contiguous().data_ptr<float>()
```

Comment:
Wait, this doesn't seem safe.  If bias is non-contiguous, won't this contiguous call allocate a new tensor, which gets freed right after the end of this line?

Reply:
My reading of the standard is that this usage is safe.  From [Temporary Object Lifetime](https://en.cppreference.com/w/cpp/language/lifetime#Temporary_object_lifetime):

> All temporary objects are destroyed as the last step in evaluating the full-expression that (lexically) contains the point where they were created, and if multiple temporary objects were created, they are destroyed in the order opposite to the order of creation. This is true even if that evaluation ends in throwing an exception.

Where a full-expression according to the [C++ standard [draft]](https://timsong-cpp.github.io/cppwp/n3337/intro.execution#10) is defined as:

> A full-expression is an expression that is not a subexpression of another expression.

Stack Overflow question: https://stackoverflow.com/a/6059134
Example: http://cpp.sh/9tn6d

I can pull this out and convert to below as well:

```
if (bias && bias->defined()) {
  bias = bias->contiguous();
}
```

Reply:
Wait, does xnn_create_convolution2d_nhwc_f32 copy the bias?  If so, then you're definitely right.  For some reason I was assuming it retained a pointer to the original bias data.